### DISCUSSION TITLE
Unable to build my project: MGCB command "exited with code 1" at compile time

### DISCUSSION THREAD

**Discussion by QuQuasar** [MonoGame/MonoGame#8608] [General]
I have recently moved to a new PC and upgraded my project from 3.8.1 to 3.8.2 in the process. 

Note: My project requires some minor modifications to MonoGame, so I build from source. I installed it by following the instructions in the readme (clone the source, set up the submodules, open and build Framework.DesktopGL.sln, open and build Tools.Windows.sln). I also installed the project templates VS extension.

This is the error I'm seeing:

`The command "dotnet  /quiet /@:"C:\Users\Quasar\source\repos\Species\Species\Content\Content.mgcb" /platform:DesktopGL /outputDir:"C:/Users/Quasar/source/repos/Species/Species/Content/bin/DesktopGL/Content" /intermediateDir:"C:/Users/Quasar/source/repos/Species/Species/Content/obj/DesktopGL/net8.0/Content" /workingDir:"C:/Users/Quasar/source/repos/Species/Species/Content/"" exited with code 1.`

This seems to be a problem with **$(MGCBCommand)** in MonoGame.Content.Builder.Task.targets. As you can see in the above error, it's empty.

`Command="$(DotnetCommand) $(MGCBCommand) $(MonoGameMGCBAdditionalArguments) /@:&quot;%(ContentReference.FullPath)&quot; /platform:$(MonoGamePlatform) /outputDir:&quot;%(ContentReference.ContentOutputDir)&quot; /intermediateDir:&quot;%(ContentReference.ContentIntermediateOutputDir)&quot; /workingDir:&quot;%(ContentReference.FullDir)&quot;"`

If I manually compile my assets using the mgcb editor I built from the source folder, they work fine.

 
![image](https://github.com/user-attachments/assets/f99c1789-f20c-4c57-8c5f-79d87c390d76)

I have tried [Clearing all NuGet Storage](https://github.com/MonoGame/MonoGame/discussions/8487), and [reinstalling MonoGame Extension in Visual Studio and clearing the NuGet cache](https://github.com/MonoGame/MonoGame/discussions/8434), but the problem persists.

Can anyone help? At the very least, can someone let me know where **$(MGCBCommand)** is defined?

**Comment by jonathanmccarthy** [comment]
You are in luck, I moved to a new PC last week and had the same issue.
I also had the issue when I got my last PC in 2017 so went digging for the solution.
Solution here:
https://community.monogame.net/t/resolving-mgcb-editor-not-found-error/19419

There's a bit in there where it says you need a subfolder .config containing a file called dotnet-tools.json

It solved it for me.

**Reply by QuQuasar** [reply]
I appreciate the reply, but I'm afraid that following those instructions has not solved the issue on my end. It's still trying to run the same `dotnet  /quiet` command as above, with **$(MGCBCommand)** blank (as opposed to `dotnet mgcb /quiet`, which is what I'd expect from the targets file).

I think your fix did solve _something_, though. Prior to this I wasn't able to run `dotnet mgcb /quiet` from the terminal, which struck me as odd since I could run `mgcb /quiet` without any problem. Now both run fine. 

(Edit) I may have resolved it. See below.

**Comment by QuQuasar** [comment]
I think I've resolved it.

I found the '$(MGCBPath)' property in my csproj file and realised that it was probably what was being used by the targets file in prior versions. This was likely a result of me trying to continue to use the existing monogame project instead of creating a new 3.8.2 project and copying all my files into it.

That being the case, I need to either define '$(MGCBCommand)' in the csproj file or switch out the targets file to utilize '$(MGCBPath)'. For now, I've chosen the former.

**Comment by jhoney92021** [comment]
I had a similar issue after creating a new game from a monoGame template and trying to upgrade everything to .net 10.
I found after doing a dotnet restore, and then going in and manually deleting the /bin /obj directories, and then building solved it for me.

**Comment by giokhmaladze** [comment]
on windows i was having similar issue, if i ran mgcb tools/restore directly they worked but if i ran `dotnet build` those commands failed to execute, turns out build doesn't work only in bash emulator (gitbash mingw64 in my case), works fine if i execute build command from powershell...
